meerk40t/gui/toolwidgets/tooldraw.py

```diff
@@ -49,9 +49,14 @@ def event(
             self.add_point(space_pos[:2])
             self.scene.request_refresh()
             response = RESPONSE_CONSUME
-        elif event_type == "lost":
+        elif event_type == "lost" or (event_type == "key_up" and modifiers == "escape"):
             self.series = None
-            return RESPONSE_DROP
+            if self.scene.tool_active:
+                self.scene.tool_active = False
+                self.scene.request_refresh()
+                response = RESPONSE_CONSUME
+            else:
+                response = RESPONSE_CHAIN
```

Comment:
All the tools check for lost or if escape is hit directly. If they tool already isn't active, then it chains to the tool container widget.